drivers/hmis/app/graphql/mutations/update_unit_group.rb

```diff
@@ -0,0 +1,62 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Mutations
+  class UpdateUnitGroup < CleanBaseMutation
+    argument :id, ID, required: true
+    argument :input, Types::HmisSchema::UnitGroupInput, required: true
+
+    field :unit_group, Types::HmisSchema::UnitGroup, null: true
+
+    def resolve(id:, input:)
+      unit_group = Hmis::UnitGroup.viewable_by(current_user).find_by(id: id)
+
+      access_denied! unless unit_group.present?
+      access_denied! unless current_user.permissions_for?(unit_group.project, :can_manage_units)
+
+      errors = HmisErrors::Errors.new
+
+      # If name is being changed, check that it's still unique in the project.
+      # (This is also validated on the model, but we want to return a more user-friendly error message)
+      if input.name.present? && input.name != unit_group.name
+        # rubocop:disable Style/IfUnlessModifier
+        if Hmis::UnitGroup.where(project_id: unit_group.project_id, name: input.name).exists?
+          errors.add :name, :invalid, message: 'must be unique in the project'
+        end
+        # rubocop:enable Style/IfUnlessModifier
+      end
```

Comment:
```suggestion
```

Suggest moving this validation to the model instead,
https://github.com/greenriver/hmis-warehouse/blob/1c8d8d6456ff374bc860af6a20837e2ae88ac1d3/drivers/hmis/app/models/hmis/unit_group.rb#L33
can be updated to add a custom message: `validates :name, presence: true, uniqueness: { scope: :project_id, case_sensitive: false, message: 'must be unique in the project' }`

which has the same effect!